lambda/src/replace_scp.py

```diff
@@ -0,0 +1,133 @@
+"""Replace SCP.
+
+Purpose:
+    Replace DETACH_SCP_ID SCP with ATTACH_SCP_ID SCP defined in the associated env vars
+Environment Variables:
+    LOG_LEVEL: (optional): sets the level for function logging
+            valid input: critical, error, warning, info (default), debug
+    DETACH_SCP_ID: id of the policy to detach
+    ATTACH_SCP_ID: id of the new policy to attach
+    ASSUME_ROLE_NAME: Name of role to assume
+"""
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+import os
+import sys
+
+from aws_lambda_powertools import Logger
+import boto3
+
+# Standard logging config
+LOG_LEVEL = os.environ.get("LOG_LEVEL", "info")
+LOG = Logger(
+    service="new_account_replace_scp",
+    level=LOG_LEVEL,
+    stream=sys.stderr,
+    location="%(name)s.%(funcName)s:%(lineno)d",
+    timestamp="%(asctime)s.%(msecs)03dZ",
+    datefmt="%Y-%m-%dT%H:%M:%S",
+)
+
+ATTACH_SCP_ID = os.environ["ATTACH_SCP_ID"]
+DETACH_SCP_ID = os.environ["DETACH_SCP_ID"]
+
+# Get client
+org_client = boto3.client("organizations")
+
+
+class ReplaceSCPMaxPoliciesError(Exception):
+    """Replace SCP Error for Max Policies Attached."""
+
+
+class ReplaceSCPError(Exception):
+    """All Replace SCP Errors."""
+
+
+@LOG.inject_lambda_context(log_event=True)
+def lambda_handler(event, context):  # pylint: disable=unused-argument
+    """Replace scp policy lambda handler."""
+    target_id = get_target_id(event)
+    replace_scp(target_id)
+
+
+def get_new_account_id(event):
+    """Return account id for new account events."""
+    return event["detail"]["serviceEventDetails"]["createAccountStatus"]["accountId"]
+
+
+def get_invite_org_target_id(event):
+    """Return target id for invite to org events."""
+    return event["detail"]["requestParameters"]["target"]["id"]
+
+
+def get_create_org_target_id(event):
+    """Return target id for create org events."""
+    return event["detail"]["recipientAccountId"]
+
+
+def get_target_id(event):
+    """Return account id for supported events."""
+    event_name = event["detail"]["eventName"]
+    get_account_id_strategy = {
+        "CreateAccountResult": get_new_account_id,
+        "CreateOrganizationalUnit": get_create_org_target_id,
+        "InviteAccountToOrganization": get_invite_org_target_id,
+    }
+    return get_account_id_strategy[event_name](event)
```

Comment:
quibble, since it's not always an account_id...

```suggestion
    get_target_id_strategy = {
        "CreateAccountResult": get_new_account_id,
        "CreateOrganizationalUnit": get_create_org_target_id,
        "InviteAccountToOrganization": get_invite_org_target_id,
    }
    return get_target_id_strategy[event_name](event)
```